Microsoft.DotNet.Interactive/IKernel.cs

```diff
@@ -15,6 +15,8 @@ public interface IKernel : IDisposable
     {
         string Name { get; }
 
+        IKernel Parent { get; set; }
```

Comment:
This doesn't belong on the interface. The hierarchy is really an implementation detail of the `CompositeKernel`.